core_impl/src/test/java/com/google/instrumentation/trace/SpanImplTest.java

```diff
@@ -80,15 +119,38 @@ public void toSpanData_ActiveSpan() {
             SPAN_NAME,
             parentSpanId,
             true,
+            TraceParams.DEFAULT,
             startEndHandler,
             timestampConverter,
             testClock);
     Mockito.verify(startEndHandler, Mockito.times(1)).onStart(span);
+    span.addAnnotation(Annotation.fromDescription(ANNOTATION_DESCRIPTION));
+    span.addAnnotation(ANNOTATION_DESCRIPTION, attributes);
+    NetworkEvent networkEvent =
+        NetworkEvent.builder(NetworkEvent.Type.RECV, 1).setMessageSize(3).build();
+    span.addNetworkEvent(networkEvent);
+    Link link = Link.fromSpanContext(spanContext, Link.Type.CHILD);
+    span.addLink(link);
     SpanData spanData = span.toSpanData();
     assertThat(spanData.getContext()).isEqualTo(spanContext);
     assertThat(spanData.getDisplayName()).isEqualTo(SPAN_NAME);
     assertThat(spanData.getParentSpanId()).isEqualTo(parentSpanId);
     assertThat(spanData.getHasRemoteParent()).isTrue();
+    assertThat(spanData.getAnnotations().getDroppedEventsCount()).isEqualTo(0);
+    assertThat(spanData.getAnnotations().getEvents().size()).isEqualTo(2);
+    assertThat(spanData.getAnnotations().getEvents().get(0).getTimestamp()).isEqualTo(timestamp);
+    assertThat(spanData.getAnnotations().getEvents().get(0).getEvent())
+        .isEqualTo(Annotation.fromDescription(ANNOTATION_DESCRIPTION));
+    assertThat(spanData.getAnnotations().getEvents().get(1).getTimestamp()).isEqualTo(timestamp);
+    assertThat(spanData.getAnnotations().getEvents().get(1).getEvent())
+        .isEqualTo(Annotation.fromDescriptionAndAttributes(ANNOTATION_DESCRIPTION, attributes));
+    assertThat(spanData.getNetworkEvents().getDroppedEventsCount()).isEqualTo(0);
+    assertThat(spanData.getNetworkEvents().getEvents().size()).isEqualTo(1);
+    assertThat(spanData.getNetworkEvents().getEvents().get(0).getTimestamp()).isEqualTo(timestamp);
```

Comment:
Shouldn't this test increment the clock between adding events, to test that each event uses the right timestamp?